catalog/transformer/catalog-transformer-pptx/src/main/java/ddf/catalog/transformer/input/pptx/PptxInputTransformer.java

```diff
@@ -0,0 +1,243 @@
+     * is null, then a {@link NullPointerException} will be thrown.
+     *
+     * @param inputTransformer must be non-null
+     * @throws NullPointerException
+     */
+    public PptxInputTransformer(@NotNull InputTransformer inputTransformer) {
+
+        notNull(inputTransformer, "The inputTransformer parameter must be non-null");
+
+        this.inputTransformer = inputTransformer;
+    }
+
+    @Override
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+
+        if (input == null) {
+            throw new CatalogTransformerException("Cannot transform null input.");
+        }
+
+        return transformLogic(input, id);
+    }
+
+    /**
+     * This is a three step process. First, create a FileBackedOutputStream because we need to
+     * consume the stream twice. Once for the injected inputTransformer and once for Apache POI.
+     * Next, extract the metadata with the injected input transformer. And last, use Apache POI
+     * to create the thumbnail.
+     *
+     * @param input
+     * @param id
+     * @return
+     * @throws IOException
+     * @throws CatalogTransformerException
+     */
+    private Metacard transformLogic(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+
+        try (FileBackedOutputStream fileBackedOutputStream = new FileBackedOutputStream(
+                FILE_BACKED_THRESHOLD)) {
+            try {
+                int c = IOUtils.copy(input, fileBackedOutputStream);
+                LOGGER.debug("copied {} bytes from input stream to file backed output stream", c);
+            } catch (IOException e) {
+                throw new CatalogTransformerException("Could not copy bytes of content message.",
+                        e);
+            }
+
+            Metacard metacard = extractInitialMetadata(fileBackedOutputStream.asByteSource()
+                    .openStream());
+
+            extractThumbnail(metacard,
+                    fileBackedOutputStream.asByteSource()
+                            .openStream());
+
+            return metacard;
+        }
+
+    }
+
+    /**
+     * Extract the initial metadata using the injected input transformer.
+     *
+     * @param input
+     * @return
+     * @throws IOException
+     * @throws CatalogTransformerException
+     */
+    private Metacard extractInitialMetadata(InputStream input)
+            throws IOException, CatalogTransformerException {
+        return inputTransformer.transform(input);
+    }
+
+    /**
+     * SlideShowFactory.create() handles the old-style (OLE2) PPT file and the newer OOXML PPTX files. But
+     * the old-style is not well supported (it's not even a part of the normal jars). The only reason I'm
```

Comment:
I appreciate the details here, but it would be better to avoid personal commentary.  Content is good though.